middleware/csrf_test.go

```diff
@@ -0,0 +1,70 @@
+package middleware_test
+
+import (
+	"testing"
+
+	"github.com/gobuffalo/buffalo"
+	"github.com/gobuffalo/buffalo/middleware"
+	"github.com/gobuffalo/buffalo/render"
+	"github.com/markbates/willie"
+	"github.com/stretchr/testify/require"
+)
+
+type csrfForm struct {
+	AuthenticityToken string `form:"authenticity_token"`
+}
+
+func ctCSRFApp() *buffalo.App {
+	h := func(c buffalo.Context) error {
+		return c.Render(200, render.String(c.(*buffalo.DefaultContext).Data()["authenticity_token"].(string)))
+	}
+	a := buffalo.Automatic(buffalo.Options{})
+	a.GET("/csrf", middleware.EnableCSRF()(h))
+	a.POST("/csrf", middleware.EnableCSRF()(h))
```

Comment:
```go
a.GET("/csrf", middleware.CSRF(h))
a.POST("/csrf", middleware.CSRF(h))
```

When you update the `CSRF` function to be a `buffalo.MiddlewareFunc` itself, it becomes a lot easier to wrap `buffalo.Handler` functions.

You could also do the same thing with:

```go
a.Use(middleware.CSRF)
a.GET("/csrf", h)
a.POST("/csrf", h)
```